### PR TITLE
Add web spotlight entity

### PR DESCRIPTION
**Pull request by JiriLojda**
### Motivation

Which issue does this fix? Fixes #`issue number`

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

File: src/modules/sync/utils/getContentModel.ts [hunk 1]
```diff
@@ -17,19 +27,20 @@ export const readContentModelFromFolder = async (folderName: string): Promise<Fi
...
-  >;
+  ) as ReadonlyArray<ContentTypeSnippetsSyncModel>;
   const taxonomyGroups = JSON.parse(
     await fs.readFile(`${folderName}/${taxonomiesFileName}`, "utf8"),
-  ) as unknown as ReadonlyArray<
-    TaxonomySyncModel
-  >;
+  ) as ReadonlyArray<TaxonomySyncModel>;
+
+  const webSpotlight = JSON.parse(
```
Comment (IvanKiral) [inline src/modules/sync/utils/getContentModel.ts:35]: maybe we can use zod for this? 👀 If you don't want to do that in this PR, maybe we can create a task for that :)

File: tests/unit/syncModel/webSpotlightTransformer.test.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  it("correctly transforms enabled web spotlight with existing root type", () => {
+    const environmentModel = createEnvironmentModel(true, "typeId1");
+    const expectedOutput = { enabled: true, root_type: { codename: "type_codename_1" } };
+
+    const result = transformWebSpotlightModel(environmentModel);
+
+    expect(result).toEqual(expectedOutput);
+  });
+
+  it("correctly transforms enabled web spotlight with non-existing root type", () => {
```
Comment (IvanKiral) [inline tests/unit/syncModel/webSpotlightTransformer.test.ts:47]: I would rather test when the web spotlight is deactivated but root_type isn't.. This one can happen, when you deactivate the web_spotlight :)

Moreover I am not sure, whether you can have enabled web_spotlight without any type

File: tests/unit/utils/requests.test.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ describe("request utils", () => {
     });
   });
 
-  describe.only("seriallyReduce", () => {
+  describe("seriallyReduce", () => {
```
Comment (IvanKiral) [inline tests/unit/utils/requests.test.ts:29]: thanks bro

**Review by IvanKiral — CHANGES_REQUESTED** [review]

File: src/modules/sync/utils/getContentModel.ts (hunk 1 above)
Comment (JiriLojda) [reply src/modules/sync/utils/getContentModel.ts:35]: I'll create a task to use zod to validate all the input files (both for sync and import)

Comment (JiriLojda) [reply src/modules/sync/utils/getContentModel.ts:35]: https://kontent-ai.atlassian.net/browse/DEVREL-1202

File: src/modules/sync/sync.ts [hunk 4]
```diff
@@ -43,14 +43,17 @@ export const sync = async (client: ManagementClient, diff: DiffModel, logOptions
   logInfo(logOptions, "standard", "Updating content types and adding their references");
   await updateContentTypesAndAddReferences(client, diff.contentTypes);
 
+  logInfo(logOptions, "standard", "Updating web spotlight");
```
[the remark this answers is not in the document]
Comment (JiriLojda) [reply src/modules/sync/sync.ts:46]: I am not sure what you mean

[the remark this answers is not in the document]
Comment (IvanKiral) [reply src/modules/sync/sync.ts:46]: I mean explaining the order of these operations. As we first create all the types, when enabling Web Spotlight we don't have to worry that the type we are referencing won't exist. Due to this we always use the option when we provide the `root_type` parameter. We never use `null` when enabling Web Spotlight. Ideally, we would also update the diagram in the images :D

**Review by IvanKiral — CHANGES_REQUESTED** [review]

**Review by IvanKiral — APPROVED** [review]

**Pull request merged**